course_info/views.py

```diff
@@ -170,6 +187,33 @@ def editor(request):
     course_context = _course_context(request, _ORDERED_FIELD_NAMES, True,
                                      course_instance_id=course_instance_id)
     course_context['launch_presentation_return_url'] = \
-            request.POST.get('launch_presentation_return_url')
+        request.POST.get('launch_presentation_return_url')
     course_context['textarea_fields'] = ['description', 'notes']
     return render(request, 'course_info/editor.html', course_context)
+
+
+def sort_and_format_instructor_display(course_instructor_list):
+    """
+    Returns a sorted  string  sorted by  the instructors by role_id, seniority sort, last name
+    and then formatted such that it appears as a comma delimited String with an 'and' before that last user.
+    # Eg: User1, User2 and User3.
+    """
+    instructor_display = ''
+    # Note:  when seniority_sort is null, it was getting precedence over 1(eg: null, 1, 2).  So in such cases,
+    # it is being set to a large number (picked 100) so it is lower in the sorting order. [1, 2, ...null(set to 100)]
+    course_instructor_list.sort(key=lambda x: (x.get('role')['role_id'],
```

Comment:
If you're really worried about the entries in `course_instructor_list` missing data, you probably want to be paranoid about both levels of the dictionary here, no?  If `x` is missing `role`, then the `['role_id']` item access will get you a `TypeError`.  (the `TypeError: 'NoneType' object has no attribute '__getitem__'` ones we all know and love).

You can still get what you want in a single line by remembering that you can pass a fallback return value to `.get`, and don't have to rely on the default of `None`.

```key=lambda x: (x.get('role', {}).get('role_id'),```